fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/config/FeignConfig.java

```diff
@@ -49,7 +51,8 @@ private void fillSecurityHeaders(RequestTemplate requestTemplate) {
 
     private String calculateSignature(Request request, Instant instant) {
         String xRequestId = request.headers().get(X_REQUEST_ID).stream().findFirst().orElse(null);
-        DataToSign dataToSign = new DataToSign(UUID.fromString(xRequestId), instant);
+        String operationType = request.headers().get(X_OPERATION_TYPE).stream().findFirst().orElse(null);
+        DataToSign dataToSign = new DataToSign(UUID.fromString(xRequestId), instant, OperationType.valueOf(operationType));
```

Comment:
You need to:
1. From the request, path deduce OperationType (probably some map in application.yml)
2. Check that OperationType from request equals to OperationType you calculated from pt. 1
3. And only after compute the signature
In essence, OperationType should identify endpoint (or endpoint 'class')

Reply:
Validation filter was added